homeassistant/components/wyoming/config_flow.py

```diff
@@ -54,3 +58,36 @@ async def async_step_user(
         name = asr_installed[0].name
 
         return self.async_create_entry(title=name, data=user_input)
+
+    async def async_step_hassio(self, discovery_info: HassioServiceInfo) -> FlowResult:
+        """Handle Supervisor add-on discovery."""
+        await self.async_set_unique_id(discovery_info.slug)
+        self._abort_if_unique_id_configured()
+
+        self._hassio_discovery = discovery_info
+        return await self.async_step_hassio_confirm()
+
+    async def async_step_hassio_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Confirm Supervisor discovery."""
+        errors: dict[str, str] = {}
+
+        if user_input is not None:
+            uri = urlparse(self._hassio_discovery.config["uri"])
+            if service := await WyomingService.create(uri.hostname, uri.port):
+                if not any(asr for asr in service.info.asr if asr.installed):
```

Comment:
I've added a click-up for me to turn this check into a method on `WyomingService` since it will need to include tts and conversation in the future.